frontend/src/metabase/dashboard/containers/CreateDashboardForm.tsx

```diff
@@ -0,0 +1,134 @@
+import FormProvider from "metabase/core/components/FormProvider";
+import FormInput from "metabase/core/components/FormInput";
+import FormTextArea from "metabase/core/components/FormTextArea";
+import FormSubmitButton from "metabase/core/components/FormSubmitButton";
+import FormErrorMessage from "metabase/core/components/FormErrorMessage";
+import FormCollectionPicker from "metabase/core/components/FormCollectionPicker";
+
+import * as Errors from "metabase/core/utils/errors";
+
+import Collections from "metabase/entities/collections";
+import Dashboards from "metabase/entities/dashboards";
+
+import type { CollectionId, Dashboard } from "metabase-types/api";
+import type { State } from "metabase-types/store";
+
+const DASHBOARD_SCHEMA = Yup.object({
+  name: Yup.string().required(Errors.required).max(100, Errors.maxLength),
+  description: Yup.string().nullable().max(255, Errors.maxLength),
+  collection_id: Yup.number().nullable(),
+});
+
+interface CreateDashboardProperties {
+  name: string;
+  description: string | null;
+  collection_id: CollectionId;
+}
+
+export interface CreateDashboardFormOwnProps {
+  collectionId?: CollectionId; // can be used by `getInitialCollectionId`
+  onCreate?: (dashboard: Dashboard) => void;
+  onCancel?: () => void;
+}
+
+interface CreateDashboardFormStateProps {
+  initialCollectionId: CollectionId;
+}
+
+interface CreateDashboardFormDispatchProps {
+  handleCreateDashboard: (
+    dashboard: CreateDashboardProperties,
+  ) => Promise<Dashboard>;
+}
+
+type Props = CreateDashboardFormOwnProps &
+  CreateDashboardFormStateProps &
+  CreateDashboardFormDispatchProps;
+
+function mapStateToProps(state: State, props: CreateDashboardFormOwnProps) {
+  return {
+    initialCollectionId: Collections.selectors.getInitialCollectionId(
+      state,
+      props,
+    ),
+  };
+}
+
+const mapDispatchToProps = {
+  handleCreateDashboard: Dashboards.actions.create,
+};
+
+function CreateDashboardForm({
+  initialCollectionId,
+  handleCreateDashboard,
+  onCreate,
+  onCancel,
+}: Props) {
+  const initialValues = useMemo(
+    () => ({
+      name: "",
+      description: null,
+      collection_id: initialCollectionId,
+    }),
+    [initialCollectionId],
+  );
+
+  const handleCreate = useCallback(
+    async (values: CreateDashboardProperties) => {
+      const action = await handleCreateDashboard(values);
+      const dashboard = Dashboards.HACK_getObjectFromAction(action);
+      onCreate?.(dashboard);
+    },
+    [handleCreateDashboard, onCreate],
+  );
+
+  return (
+    <FormProvider<CreateDashboardProperties>
+      initialValues={initialValues}
+      validationSchema={DASHBOARD_SCHEMA}
+      onSubmit={handleCreate}
+    >
+      {({ dirty }) => (
+        <Form>
+          <FormInput
+            name="name"
+            title={t`Name`}
+            placeholder={t`What is the name of your dashboard?`}
+            autoFocus
+          />
+          <FormTextArea
```

Comment:
Should it be `nullable`? This would automatically convert empty description to `null`